fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4

```diff
@@ -613,6 +614,10 @@ nonReserved
     | LOGICAL
     | MACRO
     | MAP
+    | MATCH
+    | MATCH_ANY
+    | MATCH_ALL
+    | MATCH_PHRASE
```

Comment:
in legacy planner, all match related keywords are reserved keywords, so remove them from nonReserved

Reply:
I will update it